gluten-celeborn/src/main/java/org/apache/spark/shuffle/gluten/celeborn/CelebornShuffleManager.java

```diff
@@ -160,33 +173,40 @@ private void initializeLifecycleManager() {
     }
   }
 
+  private <K, V, C> ShuffleHandle registerCelebornShuffleHandle(
+      int shuffleId, ShuffleDependency<K, V, C> dependency) {
+    return new CelebornShuffleHandle<>(
+        appUniqueId,
+        lifecycleManager.getHost(),
+        lifecycleManager.getPort(),
+        lifecycleManager.getUserIdentifier(),
+        shuffleId,
+        dependency.rdd().getNumPartitions(),
+        dependency);
+  }
+
   @Override
   public <K, V, C> ShuffleHandle registerShuffle(
       int shuffleId, ShuffleDependency<K, V, C> dependency) {
+    appUniqueId = SparkUtils.appUniqueId(dependency.rdd().context());
+    initializeLifecycleManager();
     // Note: generate app unique id at driver side, make sure dependency.rdd.context
     // is the same SparkContext among different shuffleIds.
     // This method may be called many times.
     if (dependency instanceof ColumnarShuffleDependency) {
-      appUniqueId = SparkUtils.appUniqueId(dependency.rdd().context());
-      initializeLifecycleManager();
-
       if (fallbackPolicyRunner.applyAllFallbackPolicy(
           lifecycleManager, dependency.partitioner().numPartitions())) {
         logger.warn("Fallback to ColumnarShuffleManager!");
         columnarShuffleIds.add(shuffleId);
         return columnarShuffleManager().registerShuffle(shuffleId, dependency);
       } else {
```

Comment:
Adding concrete types here will make the logic clearer. ```else if (h.dependency() instanceof ShuffleDependency)```, and if none match, an exception is thrown in the else.